cluster-operator/controllers/vmc/update_rancher_rolebinding.go

```
@@ -0,0 +1,95 @@
+// Copyright (c) 2022, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package vmc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/verrazzano/verrazzano/cluster-operator/apis/clusters/v1alpha1"
+	vzconst "github.com/verrazzano/verrazzano/pkg/constants"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	APIGroupRancherManagement        = "management.cattle.io"
+	APIGroupVersionRancherManagement = "v3"
+	ClusterRoleTemplateBindingKind   = "ClusterRoleTemplateBinding"
+	UserListKind                     = "UserList"
+	UserKind                         = "User"
+
+	ClusterRoleTemplateBindingAttributeClusterName      = "clusterName"
+	ClusterRoleTemplateBindingAttributeUserName         = "userName"
+	ClusterRoleTemplateBindingAttributeRoleTemplateName = "roleTemplateName"
+
+	UserUsernameAttribute = "username"
+)
+
+// updateRancherClusterRoleBindingTemplate creates a new ClusterRoleBindingTemplate for the given VMC
+// to grant the Verrazzano cluster user the correct permissions on the managed cluster
+func (r *VerrazzanoManagedClusterReconciler) updateRancherClusterRoleBindingTemplate(vmc *v1alpha1.VerrazzanoManagedCluster) error {
+	if vmc == nil {
+		r.log.Debugf("Empty VMC, no ClusterRoleBindingTemplate created")
+		return nil
+	}
+
+	clusterID := vmc.Status.RancherRegistration.ClusterID
+	if len(clusterID) == 0 {
+		r.log.Debugf("Waiting to create ClusterRoleBindingTemplate for cluster %s, Rancher ClusterID not found in the VMC status", vmc.GetName())
+		return nil
```

Does this need to return an error so that it gets retried?

---

I don't believe so, because the cluster ID getting updated should re-trigger a reconcile